src/pip/_internal/commands/uninstall.py

```diff
@@ -90,6 +93,10 @@ def run(self, options: Values, args: List[str]) -> int:
                 f'"pip help {self.name}")'
             )
 
+        externally_managed_error = get_externally_managed_error()
+        if externally_managed_error is not None:
+            raise InstallationError(externally_managed_error)
```

Comment:
I'd prefer if we raise a DiagnosticPipError subclass here.

Reply:
The None case means “not externally managed” so it should not raise an error, but otherwise sure.

Reply:
I also meant more of keeping the default string in the error as well FWIW, and bubbling the exception out of the function being called.

Reply:
Do you mean something like this?

```python
def check_current_environment_is_externally_managed():
    message: str | None = _read_externally_managed_config()
    if message is not None:
        raise ExternallyManagedError(message)
```

This looks good, but the exception class would then take `str` instead of `str | None`, so I’m not quite sure if I’m understanding you correctly.

Reply:
Tried a change; does this look better?